app/org/maproulette/framework/mixins/SearchParametersMixin.scala

```diff
@@ -18,45 +18,209 @@ import scala.collection.mutable.ListBuffer
 
 trait SearchParametersMixin {
 
+  /**
+    * Sets up a where clause searching by given SearchParameters
+    */
+  def filterOnSearchParameters(
+      params: SearchParameters
+  )(implicit projectSearch: Boolean = true): Query = {
+    var newQuery = Query.empty
+
+    newQuery = newQuery.addFilterGroup(this.filterLocation(params))
```

Comment:
You should probably instead of doing this addFilterGroup over and over again, you should just build the query from scratch like this:
```
val filterList = List(
  this.filterLocation(params),
  this.filterBounding(params),
  ...
)
Query(Filter(filterList))
```
Something like that anyway.